lib/src/open_food_api_client.dart

```diff
@@ -1224,8 +1224,11 @@ class OpenFoodAPIClient {
         body:
             'An email with a link to reset your password has been sent to the e-mail address associated with your account.',
       );
-    } else {
+    } else if (status.status is int && status.status != 500) {
       return status.copyWith(status: 400);
+    } else {
+      /// Trigger real 5xx errors
+      return status;
```

Comment:
Say the status is 502 then this case would never trigger. Everything in the besides 500 Returns a 400

Is that your wanted behavior

Reply:
Oops my bad, I created this PR while discussing and clearly that was not a good idea.